### DISCUSSION TITLE
Usage of varvalue Type

### DISCUSSION THREAD

**Discussion by Lafatalite** [robotraconteur/robotraconteur#272] [Q&A]
I want to pass a value in my project that can either be of type `int32` or `bool` .I have defined this value in the robdef file as varvalue, like this: `field varvalue data.` Then, when I tried to simulate updating this value on the server side, I encountered an error when assigning the value. The error is: `RobotRaconteur.RobotRaconteurPythonError.DataTypeException: RobotRaconteur.DataTypeError Invalid VarValue object for field "value.device[0].channel[0].data" expected Robot Raconteur type "varvalue"` I tried assigning the value like this: `self.data = 1234` and also tried `self.data = RR.VarValue(1234,Int32)` ,but both resulted in the error mentioned above.

**Comment by johnwason** [comment]
The "datatype" field of the `VarValue` is a string of the data type formatted as in a Service Definition. Try:

```
val = RR.VarValue(1234, "int32")
```

There are examples of VarValue being used in the unit tests. Search for "VarValue" in `test/python`.

There is a new getting started guide you may not have seen: https://robotraconteur.github.io/robotraconteur/doc/core/latest/getting_started/

Also there is a new mailing list: https://lists.robotraconteur.com/lists/
